### DISCUSSION TITLE
[Plugin Request] Consider creating golang templates similar to gqlgen

### DISCUSSION THREAD

**Discussion by shah** [dotansimha/graphql-code-generator#298] [General]
/vektah/gqlgen is a library for quickly creating strictly typed graphql servers in Google Go. I wonder if the GraphQL Code Generator team could use this as a starting point for creating a Go template tied to PostgreSQL database? Or Go tied to dGraph.io?

**Comment by steebchen** [comment]
Just for your information: I am currently writing a plugin for this. However, I will focus on operations (for the client) because I need it and not on resolvers (for the server). It's currently a prototype and the code is a bit hacky (most is copied from the TS plugins and then adapted) so I'm not sure if I ever merge this here, but at some point it'll be at least possible to use as an external plugin.

**Comment by shah** [comment]
@steebchen I've been using [gqlgen ](https://gqlgen.com/) for Go this year and it's worked quite well. Check it out as you start your journey.

**Comment by steebchen** [comment]
I use `gqlgen` as well, but it generates GraphQL servers; I need something for my client with query strings, frontend types and in the best case a built-in client for 100% type safety. Although it may be possible to extend gqlgen's functionality for documents and queries instead of using this, but it seems simpler to do it with this plugin (for now).

**Comment by shah** [comment]
Got it, thanks for the clarification - I've run into the same problem before and ended up hand writing my clients but your idea is better!

**Comment by prm-dan** [comment]
Hi!  Any updates a golang plugin?  I'd use it.

**Reply by dotansimha** [reply]
No update on our end, mostly because we are not golang developers, so we don't have the knowledge of creating the ideal plugin for this platform. PRs are welcome!

**Comment by bregydoc** [comment]
Today I found this project: [https://github.com/Yamashou/gqlgenc](https://github.com/Yamashou/gqlgenc)
